src/ui/components/ExplorerHome/ExplorerTimeline.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+import React, {type Node as ReactNode} from "react";
+import {extent} from "d3-array";
+import {scaleLinear} from "d3-scale";
+import {line} from "d3-shape";
+import {type Grain} from "../../../core/ledger/grain";
+
+const legendTextStyles = {
+  fontSize: "10px",
+  fill: "#fff",
+};
+const CRED_COLOR = "#6174CC";
+const GRAIN_COLOR = "#FFAA3D";
+type ExplorerTimelineProps = {|
+  +timelines: {
+    cred: Array<number>,
+    grain?: Array<Grain>,
+  },
+  +width?: number,
+  +height?: number,
+  +hasLegend?: boolean,
+  +responsive?: boolean,
+|};
+const ExplorerTimeline = (props: ExplorerTimelineProps): ReactNode => {
+  if (props.timelines.cred == null && props.timelines.grain == null) {
+    return <></>;
+  }
+  const grainExists = props.timelines.grain && true;
+  const width = props.width || 300;
+  const height = props.height || 25;
+  const viewBox = `0 0 ${width} ${height}`;
+  const intervals = props.timelines.cred.length;
+  let grainAsNumber, range, grainValues;
+  if (grainExists) {
+    const grain = props.timelines.grain || [];
+    grainAsNumber = grain.map((g) => {
+      return Number(g);
+    });
+    const credAndGrain = props.timelines.cred.concat(grainAsNumber);
+    range = extent(credAndGrain);
+    // This is a temporary/quick fix to ensure that the timeline lines fill the entire x-axis
+    // and ensures that a line is displayed when there is only one interval.
+    grainValues = grainAsNumber.concat(grainAsNumber[intervals - 1]);
+  } else {
+    range = extent(props.timelines.cred);
+  }
+  // This is a temporary/quick fix to ensure that the timeline lines fill the entire x-axis
```

Comment:
Kind of confused how added another element to the array makes it fill the entire x-axis. Shouldn't the x-axis expand with the array if you do this?

Reply:
You are correct! I think there is something weird happening with how we're using D3 here, because the line should take up the entire width of the x-axis, but for some reason it wasn't.

Reply:
the issue we opened should resolve this